tests/test_atomicity.py

```diff
@@ -0,0 +1,58 @@
+import os
+import subprocess
+import sys
+
+import pytest
+import sqlalchemy as sa
+
+from datachain.sql.types import Float32
+
+tests_dir = os.path.dirname(os.path.abspath(__file__))
+
+python_exc = sys.executable or "python3"
+
+E2E_STEP_TIMEOUT_SEC = 90
+
+
+@pytest.mark.e2e
+@pytest.mark.xdist_group(name="tmpfile")
+def test_atomicity_feature_file(tmp_dir, catalog_tmpfile):
```

Comment:
can we test this w/o creating a subprocess please?

use Session as a context for example ... (we should try to use these heavy tests as a measure of last resort if possible)

Reply:
The except hook cannot be tested outside of subprocess because it applies to the python script in general and I wanted to test if it works even when the uncatched exception is handled properly.